docs/conceptual/runtime/weight.md

```diff
@@ -0,0 +1,158 @@
+types. This is used internally by the System and Executive modules; you probably won't use it.
+
+Both `ClassifyDispatch` and `WeightData` are generic over `T`, which gets resolved into the tuple of
+all dispatch arguments except for the origin. To demonstrate, we will craft a struct that calculates
+the weight as `m * len(args)` where `m` is a given multiplier and `args` is the concatenated tuple
+of all dispatch arguments. Further, the dispatch class is `Operational` if the transaction has more 
+than 100 bytes of length in arguments.
+
+```rust
+use coded::Encode;
+use sr_primitives::weights::{DispatchClass, ClassifyDispatch, WeightData}
+
+// self.0 is the multiplier, `m`
+struct LenWeight(u32);
+
+// We don't quite know what T is. After all, different dispatches have different arguments, hence
+// `T` will be different. All that we care about is that `T` is encodable. That is always true by
+// definition. All dispatch arguments are encodable.
+impl<T: Encode> WeighData<T> for LenWeight {
+    fn weigh_data(&self, target: T) -> Weight {
+        let multiplier = self.0;
+        let encoded_len = target.encode().len() as u32;
+        multiplier * encoded_len
+    }
+}
+
+impl<T: Encode> ClassifyDispatch<T> for LenWeight {
+    fn classify_dispatch(&self, target: T) -> DispatchClass {
+        let encoded_len = target.encode().len() as u32;
+        if encoded_len > 100 {
+            DispatchClass::Operational
+        } else {
+            DispatchClass::Normal
+        }
+    }
+}
+```
+
+A weight calculator function can also be coerced to the final type of the argument, instead of 
+defining it as a vague type that is encodable. `srml-example` contains an example of how to do 
+this. Just note that, in that case, your code would roughly look like:
+
+```rust
+struct CustomWeight;
+impl WeighData<(&u32, &u64)> for CustomWeight {
+    fn weigh_data(&self, target: (&u32, &u64)) -> Weight {
+        ...
+    }
+}
+
+// given dispatch:
+decl_module! {
+    fn foo(a: u32, b: u64) { ... }
+}
+```
+
+This means that `CustomWeight` can only be used in conjunction with a dispatch with a particular
+signature `(u32, u64)`, as opposed to `LenWeight`, which can be used with anything because they 
+don't make any strict assumptions about `<T>`.
+
+## Frequently Asked Questions
+
+> _Some operations get more expensive as storage grows and shrinks. How can the weight system
+> represent that?_
+
+As mentioned, weight should be known _readily_ and _ahead of dispatch_. Peeking the storage and
+analyzing it just to determine how much weight might be consumed does not fit this definition quite
+well. In this case, the implementation of the dispatch should take the state of the change into
+account and manually take extra fees or bonds or take any other measures to make sure that the
+transaction is safe.
+
+> _Is the weighing function open to change through governance or any sort of a runtime upgrade?_
```

Comment:
Actually this was correct originally. [Weighting](https://en.wikipedia.org/wiki/Weighting) means exactly this: to calculate weights.
```suggestion
> _Is the weighting function open to change through governance or any sort of a runtime upgrade?_
```